effekt/shared/src/main/scala/effekt/RecursiveDescent.scala

```diff
@@ -711,6 +724,23 @@ class RecursiveDescent(positions: Positions, tokens: Seq[Token], source: Source)
     nonterminal:
       (`def` ~> idRef()) ~ paramsOpt() ~ maybeReturnAnnotation() ~ (`=` ~> stmt()) match {
         case id ~ (tps, vps, bps) ~ ret ~ body =>
+         if (isSemi) {
+           semi()
+
+           val startPosition = position
+
+           if (!peek(`}`) && !peek(`def`) && !peek(EOF)) {
+              // consume until the next `def` or `}` or EOF
+              while (!peek(`}`) && !peek(`def`) && !peek(EOF)) {
+                next()
+              }
+
+              val endPosition = position
+              val msg = "Unexpected tokens after operation definition. Expected either a new operation definition or the end of the implementation."
```

Comment:
Ideally, we would have some hint with a quick fix like "did you mean to do `def op() = { ... }`?", but we don't have the infrastructure for this.